### PR TITLE
update fastfold_data_workflow

### PR DESCRIPTION
**Pull request by Fazziekey**
- slove the problem of ray init on the cluster
- optimize the code of workflow

### CODE REVIEW

File: fastfold/workflow/template/fastfold_data_workflow.py [hunk 1]
```diff
@@ -89,66 +143,33 @@ def run(self, fasta_path: str, output_dir: str, alignment_dir: str=None) -> None
...
-                "binary_path": self.db_map["hhblits"]["binary"],
-                "databases": self.db_map["hhblits"]["dbs"],
-                "n_cpu": self.no_cpus,
-            }
-            hhb_fac = HHBlitsFactory(config=hhb_config)
-            # set HHBlits output path
             bfd_out_path = os.path.join(alignment_dir, "bfd_uniclust_hits.a3m")
             # generate workflow for STEP4
-            hhb_node = hhb_fac.gen_node(fasta_path, bfd_out_path)
+            bdf_node = self.hhblits_bfd_factory.gen_node(fasta_path, bfd_out_path)
```
Comment (Gy-Lu) [inline fastfold/workflow/template/fastfold_data_workflow.py:164]: bfd_node, the same below

Comment (Fazziekey) [reply fastfold/workflow/template/fastfold_data_workflow.py:164]: it is ok, there is only one bdf_node

Comment (Fazziekey) [reply fastfold/workflow/template/fastfold_data_workflow.py:164]: it depend on if used small bdf

Comment (Gy-Lu) [reply fastfold/workflow/template/fastfold_data_workflow.py:164]: no, I said typo. we have nothing to do with `bdf`, we are using `bfd` database.

**Pull request merged**